### PR TITLE
Feature/statsd  0.3.0

### PR DESCRIPTION
**Pull request by lmachetel**
Add systemd management
molecule 2.12.1
pytest 3.5.1
tox 3.0.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,84 +38,162 @@ and use:
...
+service_systemd_conf_files:
+  journal:
+    name: 'journald.conf.j2'
+    dest: '/etc/systemd/journald.conf'
+    mode: '0644'
+    owner: 'root'
+    group: 'root'
+statsd_service_systemd:
+  - dest: '/etc/systemd/system/statsd.service'
+    handler: 'HANDLER | Restart statsd'
```
Comment (achaussier) [inline README.md]: Really useful to have handler name ? This is the same handler for init and systemd services

File: README.md [hunk 2]
```diff
@@ -37,84 +38,162 @@ and use:
...
+        RestartSec: '10s'
+      Install:
+        WantedBy: 'multi-user.target'
+
+# Statsd initd specific settings
+is_initd_managed_system: "{{ _is_initd_managed_system | default(False) }}"
+statsd_service_initd:
+  - src: "{{ role_path }}/templates/init.d.j2"
+    dest: '/etc/init.d/statsd'
+    handler: 'HANDLER | Restart statsd'
```
Comment (achaussier) [inline README.md]: Really useful to have handler name ? This is the same handler for init and systemd services

File: README.md [hunk 3]
```diff
@@ -37,84 +38,162 @@ and use:
...
-  deleteIdleStats: False
+  dumpMessages: 'False'
+  deleteIdleStats: 'False'
   prefixStats: 'statsd'
 ```
 
 ## Dependencies
 
-None
+  - geerlingguy.nodejs
```
Comment (achaussier) [inline README.md]: Add a note about conditional dependency

File: defaults/main.yml [hunk 4]
```diff
@@ -28,10 +31,48 @@ statsd_nodejs_binary: '/usr/bin/nodejs'
...
+service_systemd_conf_files:
+  journal:
+    name: 'journald.conf.j2'
+    dest: '/etc/systemd/journald.conf'
+    mode: '0644'
+    owner: 'root'
+    group: 'root'
+statsd_service_systemd:
+  - dest: '/etc/systemd/system/statsd.service'
+    handler: 'HANDLER | Restart statsd'
```
Comment (achaussier) [inline defaults/main.yml]: Really useful to have handler name ? This is the same handler for init and systemd services

File: defaults/main.yml [hunk 5]
```diff
@@ -28,10 +31,48 @@ statsd_nodejs_binary: '/usr/bin/nodejs'
...
+        RestartSec: '10s'
+      Install:
+        WantedBy: 'multi-user.target'
+
+# Statsd initd specific settings
+is_initd_managed_system: "{{ _is_initd_managed_system | default(False) }}"
+statsd_service_initd:
+  - src: "{{ role_path }}/templates/init.d.j2"
+    dest: '/etc/init.d/statsd'
+    handler: 'HANDLER | Restart statsd'
```
Comment (achaussier) [inline defaults/main.yml]: Really useful to have handler name ? This is the same handler for init and systemd services

File: molecule/default/tests/test_installation.py [hunk 6]
```diff
@@ -22,45 +21,71 @@ def test_user(host):
...
 
-    if item_type == 'directory':
-        assert current_item.is_directory
-    elif item_type == 'file':
-        assert current_item.is_file
+    if host.system_info.distribution not in codename:
+        pytest.skip('{} ({}) distribution not managed'.format(
+            host.system_info.distribution, host.system_info.release))
+    else:
+
```
Comment (achaussier) [inline molecule/default/tests/test_installation.py]: You can remove else statement and additional indent for the following lines.

File: templates/journald.conf.j2 [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+#ForwardToKMsg=no
+#ForwardToConsole=no
+#ForwardToWall=yes
+#TTYPath=/dev/console
+#MaxLevelStore=debug
+#MaxLevelSyslog=debug
+#MaxLevelKMsg=notice
+#MaxLevelConsole=info
+#MaxLevelWall=emerg
+
```
Comment (achaussier) [inline templates/journald.conf.j2:44]: Perhaps use the same logic than systemd.service template

File: tasks/install_git.yml [hunk 8]
```diff
@@ -20,15 +20,7 @@
...
-  notify:
-    - 'HANDLER | Restart statsd'
-
-
-- name: 'INSTALL | NODEJS | Install statsd NodeJS dependencies'
-  npm:
-    global: "{{ statsd_npm.global }}"
-    path: "{{ statsd_npm.path }}"
-    production: "{{ statsd_npm.production }}"
+  register: '_git_finished'
```
Comment (achaussier) [inline tasks/install_git.yml]: Unused register

File: templates/journald.conf.j2 (hunk 7 above)
Comment (lmachetel) [reply templates/journald.conf.j2:44]: Next time ;)

**Pull request merged**